src/Microsoft.DotNet.Build.Tasks/VersionTools/SubmitPullRequest.cs

```diff
@@ -5,20 +5,36 @@
 using Microsoft.Build.Framework;
 using Microsoft.DotNet.VersionTools.Automation;
 using Microsoft.DotNet.VersionTools.Automation.GitHubApi;
+using Microsoft.DotNet.VersionTools.Automation.VstsApi;
+using System;
 using System.Diagnostics;
 using System.Linq;
 
 namespace Microsoft.DotNet.Build.Tasks.VersionTools
 {
     public class SubmitPullRequest : BuildTask
     {
-        [Required]
         public string GitHubAuthToken { get; set; }
-        [Required]
         public string GitHubUser { get; set; }
-        [Required]
         public string GitHubEmail { get; set; }
 
+        /// <summary>
+        /// A VSTS PAT. Setting this makes VstsInstanceName and GitHubAuthor required. Setting this
+        /// property indicates that the PR will be sent to VSTS (regardless whether GitHubAuthToken
```

Comment:
What do you think about erroring out here instead if they are both set?

Reply:
From the point of view of this task, yeah, having both doesn't make sense. I had it that way at first. But when I started running the code, I saw my GitHub rate limit tick down because the upgrade itself has to fetch from dotnet/versions. Since anonymous rate limit is much less than authenticated, I changed the task to allow both even though this task doesn't use them. I have an explanation a few lines down:

> VstsPat and GitHubAuthToken may both be set: this is useful to get a larger rate limit for GitHub requests during the earlier tasks in the update process and submit the final PR to VSTS.

Looking at it again, though (and mulling over this comment)... I can see value in actually having a `PullRequestTargetService` param that lets the user specify which service they want to use very intentionally. Then error messages would be very specific, and you can't have a mistake like accidentally submitting to GitHub because VSTS creds aren't flowed properly.